File: rust/src/maze_scripts/helper_classes/historic_position.rs
use godot::engine::Time;
use godot::prelude::*;

#[derive(PartialEq, Eq, Clone)]
pub struct HistoricPosition {
    pub tile: Vector2i,
    pub time_ms: u64,
}

impl HistoricPosition {
    pub fn new(tile: Vector2i) -> Self {
        let time_ms = Time::singleton().get_ticks_msec();
        Self { tile, time_ms }
    }
}

File: godot/csharp/maze_scripts/helper_classes/HistoricPosition.cs
using Godot;

namespace MazePerformanceGrade.csharp.maze_scripts.helper_classes;

public readonly record struct HistoricPosition(Vector2I Tile, ulong TimeMs)
{
    public readonly Vector2I Tile = Tile;
    public readonly ulong TimeMs = TimeMs;

    public HistoricPosition(Vector2I tile) : this(tile, Time.GetTicksMsec())
    {
    }
}